src/foam/nanos/script/Script.js

```diff
@@ -210,29 +249,57 @@ foam.CLASS({
         }
       ],
       javaCode: `
-        ByteArrayOutputStream baos  = new ByteArrayOutputStream();
-        PrintStream           ps    = new PrintStream(baos);
-        Interpreter           shell = createInterpreter(x);
-        PM                    pm    = new PM.Builder(x).setClassType(Script.getOwnClassInfo()).setName(getId()).build();
-
-        // TODO: import common packages like foam.core.*, foam.dao.*, etc.
-        try {
-          setOutput("");
-          shell.setOut(ps);
-          shell.eval(getCode());
-        } catch (Throwable e) {
-          ps.println();
-          e.printStackTrace(ps);
-          Logger logger = (Logger) x.get("logger");
-          logger.error(e);
-        } finally {
-          pm.log(x);
+        PM pm = new PM.Builder(x).setClassType(Script.getOwnClassInfo()).setName(getId()).build();
+        Language l = getLanguage();
+        if ( l == foam.nanos.script.Language.JSHELL ) {
+          ByteArrayOutputStream baos = new ByteArrayOutputStream();
+          PrintStream ps = new PrintStream(baos);
+          String print = null;
+          try {
+            JShell jShell = (JShell) createInterpreter(x,ps);
+            BufferedReader rdr = new BufferedReader(new StringReader(getCode()));
+            List<String> l1 = new ArrayList<String>();
+            for ( String line = rdr.readLine(); line != null; line = rdr.readLine() ) {
+              l1.add(line);
+            }
+            List<String> instructionList = new InstructionPresentation(jShell).parseToInstruction(l1);
+            EvalInstruction console = new EvalInstruction(jShell, instructionList, x);
+            print = console.runEvalInstruction();
+            ps.print(print);
+          } catch (Throwable e) {
+            e.printStackTrace();
```

Comment:
remove the printStackTrace .. this is handled by the logger below.

Reply:
I prefer to keep it. if it is possible, we need to show the error to the user in the code result part. This will help the fix issues.

Reply:
The user will be able to see the log message, there is no need to print this out to the console as well.

Reply:
Talked offline and we can have this until we stream the stacktrace to the output for JShell. @hchoura .. can you open a secondary ticket to track this work?